src/zip/checksum_reader.cr

```diff
@@ -0,0 +1,32 @@
+module Zip
+  # Computes a CRC32 while reading from an underlying IO,
+  # optionally verifying the computed value against an
+  # expected one.
+  private class ChecksumReader
```

Comment:
Do we want this class to be private and scoped in the `Zip` module? As far as I understand, CRC32 is quite useful in many other applications, so maybe we could move it to `Zlib` and make it public. What do you think?